src/machines/lesson-player-machine.ts

```diff
@@ -333,6 +319,37 @@ export const playerMachine = Machine<
         //   },
         // })
       },
+      assignLessonAndViewer: assign((context, event) => {
+        const lessonAndViewerAssignmentEvents = ['LOAD', 'LOADED', 'VIEW']
+
+        if (!lessonAndViewerAssignmentEvents.includes(event.type)) return {}
+
+        const assignment: {viewer?: any; lesson?: any} = {}
+
+        const lessonData = (event as {lesson?: any})?.lesson || context.lesson
+        const viewerData = (event as {viewer?: any})?.viewer || context.viewer
+
+        if (lessonData) {
+          assignment['lesson'] = lessonData
+        }
+        if (viewerData) {
+          assignment['viewer'] = viewerData
+        }
+
+        return assignment
+      }),
+      assignLessonData: assign((_, event) => {
+        if (event.type !== 'done.invoke.fetchLessonDataService') return {}
+
+        return {lesson: event.data}
+      }),
+    },
+    services: {
+      loadLesson: () => {
+        throw new Error(
+          "The LessonPlayerMachine's interpreter must implement `loadLesson`.",
+        )
+      },
```

Comment:
The `loadLesson` service, used by the `loading` state, needs to be included in the machine setup wherever this is being interpreted.